src/testCombiner.ts

```diff
@@ -50,6 +50,48 @@ export function createImportString(imports: string[]): string {
   );
 }
 
+export function getFunctionNameFromTest(test: string) {
+  // match the block of text that comes immediately before the first '(' in the test
+  const functionDeclaration = test.match(/\S+\s*\(/);
+  if (!functionDeclaration) {
+    throw new Error(`Unable to find function declaration in test: ${test}`);
+  }
+  // match the actual function name from the match above
+  // letters, numbers, _ and $ are all valid in java identifiers
+  const testName = functionDeclaration[0].match(/[a-zA-Z0-9\$\_]+/);
+  if (!testName) {
+    throw new Error(`Unable to find function name in declaration: ${functionDeclaration[0]}`);
+  }
+  const invalidName = functionDeclaration[0].match(/[^a-zA-Z0-9\$\_\(\ )]/);
+  if (invalidName) {
+    throw new Error([
+      `Function name contains an invalid character (${invalidName[0]})`,
+      `for a Java identifier: ${functionDeclaration[0]}`].join(' '),
+      );
+  }
+  return testName[0];
+}
+
+export function deduplicateTestNames(tests: ITestData[]): ITestData[] {
+  const dedupedTests: ITestData[] = [];
+  const testNames = new Map();
+  for (const test of tests) {
```

Comment:
Maybe this since it's data from the tests? It seems strange to use `test.test` everywhere
```suggestion
  for (const data of tests) {
```

Reply:
I think test is a good name for this, but that `ITestData { test: string }` should be something like `ITestData { source: string }`.

What do others think? @peterjwest @adamkowalczyk @ryanbarwick 
Have to be a bit careful with this because it's a property of an interface.

Reply:
Once argument for keeping it as `test` is that `ITestData` is currently a subset of `TestDocument`, so they can be (and are) passed directly to `genTestClass` on Platform.

However, we shouldn't be bound by old Platform, so I'd be up for a name change, as long as we make sure all the in house repos using this library get updated.

I'd favour `body` over `source` I think.

Reply:
I could put this in as a unrelated change. If GitHub search is to be believed you're the only one currently using this, Adam.
https://github.com/diffblue/cover-client-js/blob/b22279704b1544d66a6b71aed1ef38043ee2a557/src/combiner.ts